tests/test_load.py

```diff
@@ -245,6 +246,42 @@ def test_load_dataset_local_with_default_in_memory(
     assert (dataset["train"].dataset_size < max_in_memory_dataset_size) is expected_in_memory
 
 
+class TestLoadDatasetOnlySplits:
```

Comment:
We should also check this scenario:
1. users load a dataset with the "train" split only
2. users reloads this dataset but this time asking for the "test" set only

The caching mechanism should notice that the "test" set is missing and download and prepare it. If I'm not wrong currently this would fail because of this line:

https://github.com/huggingface/datasets/blob/097129df8e3f357c24b3ad439b252e0d8b849dfa/src/datasets/builder.py#L504-L511

Instead of checking for the cache_dir, we should check for the actual arrow split file to exist inside cache_dir

Reply:
Current implementation of `load_dataset` assumes that if there is something in the cache, then it is _all_ you can get, unless you pass `force_download`.

I assumed that if the user passes a parameter to download only one split, then they are aware that only that split is in the cache and only that split can be loaded. In order to force a subsequent download of other split, they should pass `force_download`.

Reply:
Not a big fan of that. In my opinion it should detect that the requested split is missing and generate it.
This will avoid some confusions for users

Reply:
This is also a breaking change we can't afford IMO